HeterogeneousCore/AlpakaCore/README.md

```diff
@@ -0,0 +1,403 @@
+    }
+
+    std::optional<OtherProduct> produce(device::Record<TestRecord> const& iRecord) {
+      // get input data in the device memory space
+      auto const& deviceInput = iRecord.get(token_);
+
+      // allocate data product on the device memory
+      OtherProduct deviceProduct(iRecord.queue());
+
+      // run the algorithm, potentially asynchronously
+      algo_.fill(iRecord.queue(), deviceInput, deviceProduct);
+
+      // return the product without waiting
+      return std::move(deviceProduct);
+    }
+
+  private:
+    edm::ESGetToken<SimpleProduct, TestRecord> token_;
+    
+    OtherAlgo algo_;
+  };
+
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/ModuleFactory.h"
+DEFINE_FWK_EVENTSETUP_ALPAKA_MODULE(ExampleAlpakaDeviceESProducer);
+```
+
+## Configuration
+
+There are a few different options for using Alpaka-based modules in the CMSSW configuration.
+
+In all cases the configuration must load the necessary `ProcessAccelerator` objects (see below) For accelerators used in production, these are aggregated in `Configuration.StandardSequences.Accelerators_cff`. The `runTheMatrix.py` handles the loading of this `Accelerators_cff` automatically. The HLT menus also load the necessary `ProcessAccelerator`s.
+```python
+## Load explicitly
+# One ProcessAccelerator for each accelerator technology
+process.load("HeterogeneousCore.CUDACore.ProcessAcceleratorCUDA_cfi")
+
+# And one ProcessAccelerator for Alpaka
+process.load("HeterogeneousCore.AlpakaCore.ProcessAcceleratorAlpaka_cfi")
+
+## Or, just load
+process.load("Configuration.StandardSequences.Accelerators_cff")
+```
+
+### Explicit module type (non-portable)
+
+The Alpaka modules can be used in the python configuration with their explicit, full type names
+```python
+process.producerCPU = cms.EDProducer("alpaka_serial_sync::ExampleAlpakaProducer", ...)
+process.producerGPU = cms.EDProducer("alpaka_cuda_async::ExampleAlpakaProducer", ...)
+```
+Obviously this kind of configuration can be run only on machines that provide the necessary hardware. The configuration is thus explicitly non-portable.
+
+
+### SwitchProducerCUDA (semi-portable)
+
+A step towards a portable configuration is to use the `SwitchProcucer` mechanism, for which the currently the only concrete implementation is [`SwitchProducerCUDA`](../../HeterogeneousCore/CUDACore/README.md#automatic-switching-between-cpu-and-gpu-modules). The modules for different Alpaka backends still need to be specified explicitly
+```python
+from HeterogeneousCore.CUDACore.SwitchProducerCUDA import SwitchProducerCUDA
+process.producer = SwitchProcuderCUDA(
+    cpu = cms.EDProducer("alpaka_serial_sync::ExampleAlpakaProducer", ...),
+    cuda = cms.EDProducer("alpaka_cuda_async::ExampleAlpakaProducer", ...)
+)
+
+# or
+
+process.producer = SwitchProducerCUDA(
+    cpu = cms.EDAlias(testProducerCPU = cms.EDAlias.allProducts(),
+    cuda = cms.EDAlias(testProducerGPU = cms.EDAlias.allProducts()
```

Comment:
```suggestion
    cpu = cms.EDAlias(producerCPU = cms.EDAlias.allProducts(),
    cuda = cms.EDAlias(producerGPU = cms.EDAlias.allProducts()
```
(maybe; in case these names are supposed to be consistent with the previous snippet)

Reply:
Thanks, my purpose was indeed to be consistent with the previous snippet.